### PR TITLE
Fix move_to_top and insert_at methods

### PR DESCRIPTION
**Pull request by exAspArk**
Hi,

I had a case when `move_to_top` method did't work.
For example, I have elements with the following positions:

``` ruby
SimpleMixin.order("id asc").pluck(:pos)
#=> [0, 2, 3, 1]
```

When I tried to move last element from position `1` to the top by calling `move_to_top` method I got:

``` ruby
SimpleMixin.order("id asc").pluck(:pos)
#=> [1, 2, 3, 1]
```

It incremented position of higher elements, in this case element with position `0` to `1`. However, now we have 2 elements with position `1`. To fix it I'd like to make the following changes. 

I also realized that creating a record like `SimpleMixin.create(id: 1, pos: 0)` will set position `1` by default. Can we also use the same behavior with `insert_at` method? If I call `some_item.insert_at(0)` maybe it's better to set position `1` by default too? What do you think?

### CODE REVIEW

File: test/test_list.rb [hunk 1]
```diff
@@ -37,8 +37,8 @@ def rails_3
   defined?(ActiveRecord::VERSION) && ActiveRecord::VERSION::MAJOR >= 3
 end
 
-def rails_4
-  defined?(ActiveRecord::VERSION) && ActiveRecord::VERSION::MAJOR >= 4
+def rails_4_1
+  defined?(ActiveRecord::VERSION) && ActiveRecord::VERSION::MAJOR >= 4 && ActiveRecord::VERSION::MINOR >= 1
```
Comment (exAspArk) [inline test/test_list.rb]: `enums` were released with Rails 4.1

**Comment by swanandp** [comment]
Can you remove the `.ruby-version` from the commit? Ideally: Remove with a new commit, squash it against the last commit, and force push.

**Comment by exAspArk** [comment]
@swanandp yes, sure, if you want. Removed `.ruby-version` from `.gitignore` file. 
Looks like there are broken tests in `master` branch :(

**Comment by brendon** [comment]
Could not find acts_as_list-0.6.0 in any of the sources

Not sure why it's trying to bring that in when the version is now 0.7.0? Was there something I needed to change elsewhere?

**Comment by exAspArk** [comment]
@brendon I guess you should also bump the version in `*.gemfile.lock` files. 
However, tests were broken even before your commit https://travis-ci.org/swanandp/acts_as_list/builds

**Comment by brendon** [comment]
I see, it's `appraisal`! I'm just updating those gemspec's to push a new version (Again)

**Comment by swanandp** [comment]
Ah, good catch. I should've checked that before pushing.

**Comment by brendon** [comment]
Is this one still a problem?

**Comment by brendon** [comment]
Closing due to inactivity.

**Pull request closed without merging**